src/dfcx_scrapi/core/agents.py

```diff
@@ -53,6 +53,7 @@ def __init__(
             self.agent_id = agent_id
             self.client_options = self._set_region(agent_id)
 
+    @scrapi_base.api_call_counter_decorator
```

Comment:
@MRyderOC can we move this decorator to the `list_agents` method directly instead?  

Technically the `_build_list_agents_client_request` doesn't make an API call, only builds the proto object in preparation for the call. The actual call happens on line 120.

Reply:
After looking at this a bit more, I realize why you placed the decorator there. 😄   
You were right. The `list_agents` method was written very weird, so I went ahead and refactored that instead.